subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultLenientConfiguration.java

```diff
@@ -281,9 +304,26 @@ public ConfigurationInternal getConfiguration() {
         return loadTransientGraphResults(getSelectedArtifacts()).getRootNode().getPublicView().getChildren();
     }
 
-    private static class FilesAndArtifactCollectingVisitor extends ArtifactCollectingVisitor {
+    private static class LenientFilesAndArtifactResolveVisitor extends ArtifactCollectingVisitor {
         private final Set<File> files = Sets.newLinkedHashSet();
 
+        public void visitArtifact(AttributeContainer variant, ResolvedArtifact artifact) {
+            if (isExternalModuleArtifact(artifact)) {
```

Comment:
I think it is due to limiting what is inside the  `catch (ArtifactResolveException e) { /*ignore*/ }` to a minimum. But I also think we can simplify it. I do not see any harm in calling `getFile()` on all artifacts. And the catch is limited to the specific exception type.

I removed the `isExternalModuleArtifact()` check.